### PR TITLE
xdp-dump: add first set of selftests

### PR DESCRIPTION
**Pull request by chaudron**
Signed-off-by: Eelco Chaudron <echaudro@redhat.com>

### CODE REVIEW

File: xdp-dump/test-xdpdump.sh [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+#
+# shellcheck disable=2039
+#
+XDPDUMP=./xdpdump
+XDP_LOADER=../xdp-loader/xdp-loader
+ALL_TESTS="test_help test_interfaces test_capt_pcap test_capt_term test_exitentry test_snap test_none_xdp"
+
+print_result()
+{
+    echo "$RESULT"
```
Comment (tohojo) [inline xdp-dump/test-xdpdump.sh:12]: Please either declare this as a global variable, or just pass it in as a parameter?

File: xdp-dump/test-xdpdump.sh [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+    local OUTPUT_FILE="/tmp/${NS}_PID_${1}"
+    kill -SIGINT "$1"
+    sleep 1 # Wait to make sure the buffer is flushed after the shutdown
+    cat "$OUTPUT_FILE"
+    rm "$OUTPUT_FILE" >& /dev/null
+}
+
+test_help()
+{
+    local RESULT=""
```
Comment (tohojo) [inline xdp-dump/test-xdpdump.sh:41]: This breaks with the expectation of a global variable above, doesn't it?

Comment (chaudron) [reply xdp-dump/test-xdpdump.sh:41]: Yes I forgot to cleanup this one :( RESULT should be global, will make the change

File: xdp-dump/test-xdpdump.sh [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+    if ! [[ $RESULT =~ $PASS_EXIT_REGEX ]]; then
+        print_result "IPv6 exit packet not received"
+        return 1
+    fi
+
+    $XDP_LOADER unload "$NS" --all || return 1
+    $XDP_LOADER load "$NS" "$TEST_PROG_DIR/xdp_drop.o" || return 1
+
+    PID=$(start_background "$XDPDUMP -i $NS --rx-capture=exit")
+    ping6 -W 2 -c 1 "$INSIDE_IP6" # Note that this ping will fail!!
```
Comment (tohojo) [inline xdp-dump/test-xdpdump.sh:191]: Do we really need a 2-second wait for a ping we know is going to fail? Makes the test run slower than it needs to be...

Comment (chaudron) [reply xdp-dump/test-xdpdump.sh:191]: You are right, I'll update it with
"ping6 -W 0.1 -c 1 "$INSIDE_IP6" # Note that this ping will fail!!"

**Comment by chaudron** [comment]
FYI I did notice that some times ping is failing in the veth/netns, re-running the test makes it work. I could not quickly figure out what is wrong in these cases, but I suspect my vagrant fedora32 image.

**Comment by tohojo** [comment]
> FYI I did notice that some times ping is failing in the veth/netns, re-running the test makes it work. I could not quickly figure out what is wrong in these cases, but I suspect my vagrant fedora32 image.

Well you're relying on 'sleep' for timing, so I guess the tests are inherently racy. I suppose we can live with that for a first pass, though :)

**Comment by chaudron** [comment]
> > FYI I did notice that some times ping is failing in the veth/netns, re-running the test makes it work. I could not quickly figure out what is wrong in these cases, but I suspect my vagrant fedora32 image.
> 
> Well you're relying on 'sleep' for timing, so I guess the tests are inherently racy. I suppose we can live with that for a first pass, though :)

This is why there is an inial ping in some tests ` ping6 -W 2 -c 1 "$INSIDE_IP6" || return 1 ` as this even fails. This is just after you've setup the netns part. I'll blame my testup for now :)

**Pull request merged**